designs/carbon-aware.md

```diff
@@ -0,0 +1,197 @@
+
+RFC: https://github.com/aws/karpenter/issues/4630
+
+## Fundamentals of Green Software
+I will try to keep it simple. The reader should be familiar with the following.
+
+A cluster's emissions is made of two elements: embodied emissions and operational emissions. To get the total emissions, one can add them togeather.
+
+- **Embodied carbon emissions**: Manufacturing emissions (CO₂e) amortized over instance lifetime (usually 4 years) divided by how long we use the instance
+- **Operational carbon emissions**: Carbon emitted by electricity grid to produce electricity for the instance in the region where it is used, multiplied by PUE
+
+There is a lot more to Green Software. If you want to learn more, I recommend you to visit [Green Software Practitioner](https://learn.greensoftware.foundation/) (a Green Software Foundation project - an affiliate of the Linux Foundation).
+
+## Solution
+
+### Feature Gate
+The feature is proposed to be controlled using a [feature gate](https://karpenter.sh/docs/concepts/settings/#feature-gates).
+
+| **Feature** | **Default** |         **Config Key**          | **Stage** |    **Since**    | **Until** |
+| :---------: | :---------: | :-----------------------------: | :-------: | :-------------: | :-------: |
+| CarbonAware |    false    | featureGates.carbonAwareEnabled |   Alpha   | v0.32.0/v0.33.0 |           |
+
+### Carbon emissions data source
+Currently the best option is to create estimates based on the methodology used in [Boaviztapi](https://github.com/Boavizta/boaviztapi).
+
+[Try out Boaviztapi on the Datavizta demo website](https://datavizta.boavizta.org/cloudimpact).
+
+#### Licensing
+Boaviztapi is licensed under [`GNU Affero General Public License v3.0`](https://github.com/Boavizta/boaviztapi/blob/main/LICENSE). Therefore, as far as I know, we must license their data under the same license if used in the Karpenter repository.
+
+#### Limitations
+There is a discrepancy between the available instances known to Karpenter and instances know to Boaviztapi. This means that as it is right now, it is not possible to get carbon emissions data for all instances types. This is mostly the case for new instance types such as m7g. Around 290 out of 700 instance types is missing data. See full comparison in [this Gist](https://gist.github.com/JacobValdemar/e1342013c0f5c980126f6a1feb66b4a1).
+
+I will attempt to eleminate this discrepancy, but it might not be possible. It will probably not always be possible to have an updated list of estimated carbon emissions for all instances as AWS continue to release new instance types. We should consider what to do with instance types that we do not have carbon emission estimates for.
+
+Approaches to handle this:
```

Comment:
Why not exclude instances that are not emission-priced when the `carbonAwareEnabled` feature gate is enabled? As a user if I were strict about this accounting I would not want the risk of polluting my reckoning with non-determinate inputs.

Reply:
@jackfrancis I agree. I also think an exclusion is the best option in case we can't estimate the emissions accurately. The method I thought about excluding them was to assigning them an absurd high price so they will never be picked voluntarily. However, I want to improve the dataset that we depend on, so that we can use as many instance types as possible. See this issue for reference https://github.com/Boavizta/boaviztapi/issues/232

Reply:
My personal take is that if I were a customer I would prefer not to use something at all that isn't carbon-priced.

1. I would be biased to suspect that such a thing was more likely than not to be carbon-inefficient (i.e., a previous generation of tech)
2. I think that exclusion is a better forcing function for incentivizing such things to do the work of calculating cost, and/or marginalizing them from the marketplace (i.e., replacing them w/ newer, more carbon-efficient generations of tech)